### ISSUE TITLE
Annotation for declaring redirect for dynamic routes

### ISSUE THREAD

**Issue by pleku** [vaadin/flow#4802]
### Use case
When creating routes for my app, I want to define a redirect for certain routes that are not yet registered, so that I can guide the user to the correct place like login to make the route available, instead of showing 404 for a route that might be bookmarked but not yet accessible

When creating routes for my app, I want to easily define a reroute to dynamically added and removed routes when those have been registered for the current user, so that I can easily redirect the user to correct location so that they can renew their subscription for that part of the app

When creating an app, I want to statically declare a route that is dynamically registered, so that I can keep declaring the path for all routes the same way instead of splitting the logic in different places

### Acceptance Criteria
- Demo in flow-demos that highlights the feature usage (API) 
- Tutorial in routing tutorials about having a redirect for views that are not available without a login

### Implementation details
This could be done after #4799 with just having more logic in the phase when the routes are added and removed, or by implemented in application code with before-enter-listeners. But this would mean that you have to register the view eagerly, or by adding more logic to the callback that registers/unregisters the routes.

If we choose to add a feature for this, here is a very quick API Proposal just as an example:
```java
@DynamicRoute(redirect = "login")
@Route("profile")
ProfileView extends AbstractView { ... }

...

private void loginHandler(LoginEvent event) { // application code
    RouteRegistry registry = event.getUI().getRouter();
    registry.addRoute(ProfileView.class, event.getUI().getSession());
}
```

But this should be DX tested before implementation against having the logic in the registration phase, or having the logic in the before enter listener, or by having this type of feature in the annotation.

**Comment by pleku** [comment]
As we went with the approach of not using another annotation, but instead have the `registerOnStartup=false` option for `Route`, this enhancement is kind of invalid.

If someone would need this or something similar for the not-yet-registered-routes, they can create another issue for this. At the moment it is not that clear what value this would add (if any) and thus closing the issue.
